ui-admin/src/forms/designer/split/controls/FormElementOptions.tsx

```diff
@@ -27,9 +31,20 @@ export const FormElementOptions = ({
     <div className="d-flex justify-content-end">
       <div className="d-flex border rounded-3 rounded-top-0 border-top-0 bg-light">
         <IconButton icon={faCode}
-          aria-label={showJsonEditor ? 'Switch to designer' : 'Switch to JSON editor'}
+          disabled={showFreetextMode}
+          aria-label={
+            showJsonEditor ?
+              'Switch to designer' :
+              showFreetextMode ? 'You must exit the freetext editor to switch to JSON editor' :
+                'Switch to JSON editor'
+          }
           onClick={() => setShowJsonEditor(!showJsonEditor)}
         />
+        <IconButton icon={showFreetextMode ? faKeyboard : faWandMagicSparkles}
+          disabled={showJsonEditor}
+          aria-label={showFreetextMode ? 'Switch to designer' : 'Switch to freetext editor'}
+          onClick={() => setShowFreetextMode(!showFreetextMode)}
+        />
```

Comment:
There seemed to be some weirdness when I entered freetext mode in a panel where it didn't properly create a stableid and I ended up getting an error preventing me from saving.